akka-docs/rst/java/code/docs/persistence/PersistenceQueryDocTest.java

```diff
@@ -0,0 +1,380 @@
+ */
+
+package docs.persistence;
+
+import static akka.pattern.Patterns.ask;
+
+import akka.actor.*;
+import akka.dispatch.Mapper;
+import akka.event.EventStreamSpec;
+import akka.japi.Function;
+import akka.japi.Procedure;
+import akka.japi.pf.ReceiveBuilder;
+import akka.pattern.BackoffSupervisor;
+import akka.persistence.*;
+import akka.persistence.query.*;
+import akka.persistence.query.javadsl.ReadJournal;
+import akka.stream.ActorMaterializer;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.util.Timeout;
+import docs.persistence.query.MyEventsByTagPublisher;
+import docs.persistence.query.PersistenceQueryDocSpec;
+import org.reactivestreams.Subscriber;
+import scala.collection.Seq;
+import scala.collection.immutable.Vector;
+import scala.concurrent.Await;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Duration;
+import scala.concurrent.duration.FiniteDuration;
+import scala.runtime.Boxed;
+import scala.runtime.BoxedUnit;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class PersistenceQueryDocTest {
+
+  final Timeout timeout = Timeout.durationToTimeout(FiniteDuration.create(3, TimeUnit.SECONDS));
+
+  //#my-read-journal
+  class MyReadJournal implements ReadJournal {
+    private final ExtendedActorSystem system;
+
+    public MyReadJournal(ExtendedActorSystem system) {
+      this.system = system;
+    }
+
+    final FiniteDuration defaultRefreshInterval = FiniteDuration.create(3, TimeUnit.SECONDS);
+
+    @SuppressWarnings("unchecked")
+    public <T, M> Source<T, M> query(Query<T, M> q, Hint... hints) {
+      if (q instanceof EventsByTag) {
+        final EventsByTag eventsByTag = (EventsByTag) q;
+        final String tag = eventsByTag.tag();
+        long offset = eventsByTag.offset();
+
+        final Props props = MyEventsByTagPublisher.props(tag, offset, refreshInterval(hints));
+
+        return (Source<T, M>) Source.<EventEnvelope>actorPublisher(props)
+                                    .mapMaterializedValue(noMaterializedValue());
+      } else {
+        // unsuported
+        return Source.<T>failed(
+          new UnsupportedOperationException(
+            "Query $unsupported not supported by " + getClass().getName()))
+                     .mapMaterializedValue(noMaterializedValue());
+      }
+    }
+
+    private FiniteDuration refreshInterval(Hint[] hints) {
+      FiniteDuration ret = defaultRefreshInterval;
+      for (Hint hint : hints) {
+        if (hint instanceof RefreshInterval) {
+          ret = ((RefreshInterval) hint).interval();
+        }
+      }
+      return ret;
+    }
+
+    private <I, M> akka.japi.function.Function<I, M> noMaterializedValue() {
+      return param -> (M) null;
+    }
+  }
+  //#my-read-journal
+
+  void demonstrateBasicUsage() {
+    final ActorSystem system = ActorSystem.create();
```

Comment:
The reason that it's here is for it to never actually be created (those are "compile tests")